src/Ninja/G1CP/Content/Tests/test225.d

```diff
@@ -0,0 +1,42 @@
+/*
+ * #225 Mordrag can be told to "get out" in New Camp
+ *
+ * A variable and dialog are temporarily set and the condition function of the dialog is called.
+ *
+ * Expected behavior: The condition function will return FALSE.
+ */
+func int G1CP_Test_225() {
+    var int funcId; funcId = G1CP_Testsuite_CheckDialogConditionFunc("Org_826_Mordrag_HauAb_Condition");
+    var int infoId; infoId = G1CP_Testsuite_CheckInfo("Org_826_Mordrag_GotoNewcamp");
+    var C_Npc npc; npc = G1CP_Testsuite_FindNpc("Org_826_Mordrag");
+    var int aiVarId; aiVarId = G1CP_Testsuite_CheckIntConst("AIV_WASDEFEATEDBYSC", 0);
+    var int varId; varId = G1CP_Testsuite_CheckIntVar("Thorus_MordragKo", 0);
+    G1CP_Testsuite_CheckPassed();
+
+    // Backup values
+    var int toldBak; toldBak = Npc_KnowsInfo(hero, infoId);
+    var int aiVarBak; aiVarBak = G1CP_NpcGetAiVarI(npc, aiVarId, FALSE);
+    var int varBak; varBak = G1CP_GetIntVarI(varId, 0, 0);
+
+    // Set new values
+    G1CP_SetInfoToldI(infoId, TRUE);
+    G1CP_NpcSetAiVarI(npc, aiVarId, TRUE);
+    G1CP_SetIntVarI(varId, 0, 0);
+
+    // Call dialog condition function
+    G1CP_Testsuite_Call(funcId, npc, hero, FALSE);
+    var int ret; ret = MEM_PopIntResult();
```

Comment:
To cover all ends, this test is incomplete. The fix introduces *two* new or-conditions for which the dialog condition should fail, but the dialog condition is only tested here *once*, for both of them at the same time. The test should have two passes (i.e. calling the dialog condition twice), for either of the two conditions. Only that will fully cover testing the fix.

Reply:
The fix introduces two new **and**-conditions, so I don't see the problem here. Or am I mistaken?

Reply:
Yes, you are right. The dialog condition returns true if both new conditions are met, but the dialog condition returns false if either of the new conditions if false. The test currently confirms that the dialog condition returns false if the dialog "Org_826_Mordrag_GotoNewcamp" was told. What the test should also check is, that the dialog condition also returns false if only the variable "Thorus_MordragKo" is not set to `LOG_RUNNING`.

Reply:
Ah I see. Thanks for the explanation, I will update the test accordingly.

Reply:
Thanks!